packages/transcript-view/dev_tools/transcript-view-dev-options.ts

```diff
@@ -0,0 +1,316 @@
+    `;
+  }
+
+  get bottomContextHeightTemplate(): TemplateResult {
+    return html`
+      <label> Bottom Context Size:</label>
+      <input
+        type="range"
+        min="10"
+        max="100"
+        .value=${this.bottomContextHeight}
+        @input=${this.handleBottomContextSlide}
+        @change=${this.handleBottomContextSlide}
+      />
+      ${this.bottomContextHeight}px
+    `;
+  }
+
+  static get styles(): CSSResult {
+    return css`
+      .container {
+        background-color: white;
+      }
+    `;
+  }
+
+  private searchResultIndexChanged(e: CustomEvent): void {
+    const event = new CustomEvent('searchResultIndexChanged', {
+      detail: { searchResultIndex: e.detail.searchResultIndex },
+      bubbles: true,
+      composed: true,
+    });
+    this.dispatchEvent(event);
+  }
+
+  private changeAutoScroll(e: Event): void {
+    const target = e.target as HTMLFormElement;
+    this.autoScroll = target.checked;
+    const event = new CustomEvent('shouldAutoScroll', {
+      detail: { autoScroll: this.autoScroll },
+      bubbles: true,
+      composed: true,
+    });
+    this.dispatchEvent(event);
+  }
+
+  private changeShowContextZones(e: Event): void {
+    const target = e.target as HTMLFormElement;
+    this.showContextZones = target.checked;
+    const event = new CustomEvent('showContextZonesChanged', {
+      detail: { showContextZones: this.showContextZones },
+      bubbles: true,
+      composed: true,
+    });
+    this.dispatchEvent(event);
+  }
+
+  private setNormalTimerDelay(): void {
+    this.changeTimerDelay(1000);
+  }
+
+  private setFastTimerDelay(): void {
+    this.changeTimerDelay(500);
+  }
+
+  private setFasterTimerDelay(): void {
+    this.changeTimerDelay(250);
+  }
+
+  private changeTimerDelay(delay: number): void {
+    this.timerDelay = delay;
+    this.startTimer();
+  }
+
+  private startStopTimer(): void {
+    if (this.timerRunning) {
+      this.stopTimer();
+    } else {
+      this.startTimer();
+    }
+  }
+
+  private startTimer(): void {
+    this.stopTimer();
+    this.timerRunning = true;
+    this.timer = window.setInterval(() => {
+      if (this.currentTime >= 445) {
+        this.stopTimer();
+        return;
+      }
+      this.currentTime += 1;
+      this.emitCurrentTimeChangedEvent();
+    }, this.timerDelay);
+  }
+
+  private stopTimer(): void {
+    this.timerRunning = false;
+    window.clearInterval(this.timer);
+  }
+
+  handleCurrentTimeSlide(e: Event): void {
+    this.stopTimer();
+    const target = e.target as HTMLFormElement;
+    const seconds = parseFloat(target.value);
+    this.currentTime = seconds;
+    this.emitCurrentTimeChangedEvent();
+  }
+
+  emitCurrentTimeChangedEvent(): void {
+    const event = new CustomEvent('currentTimeChanged', {
+      detail: { currentTime: this.currentTime },
+      bubbles: true,
+      composed: true,
+    });
+    this.dispatchEvent(event);
+  }
+
+  handleTopContextSlide(e: Event): void {
```

Comment:
**Suggestion**: Refactor this and `handleBottomContextSlide` to call a parent method that accepts a second argument `property`, allowing you to DRY the code.

Reply:
I tried DRYing this up, but it got kind of weird with the amount of properties I ended up having to pass around. This file is just a dev options file (doesn't get used in the actual transcript view) so I don't think there's much harm in code repetition here.

Reply:
Wouldn't it be:

```javascript
handleContextSlide(direction, e: Event): void {
    const target = e.target as HTMLFormElement;
    const height = target.value;
    this.bottomContextHeight = height;
    const event = new CustomEvent(`${direction}ContextHeightChanged`, {
      detail: { height },
      bubbles: true,
      composed: true,
    });
    this.dispatchEvent(event);
}
```

and then you can have `handleBottomContextSlide` call this?

Reply:
You also need to pass around which context to resize (`this.bottomContextHeight` / `this.topContextHeight`) and I couldn't get that to work quite properly. This also opens up the question of "should there be 2 different context height change events or just one". ie:
```
const event = new CustomEvent(`contextHeightChanged`, {
  detail: { height: height, position: `top`/`bottom` },
  bubbles: true,
  composed: true,
});
```

Reply:
I think I'm going to hold off on this refactor until I have more time. This is a very small use case (only for dev options) and I don't think really needs to be elegant or anything. There's some things I need to figure out too (how to pass around which variable that needs incrementing) and also whether to generalize the event. This also means updating the demo app consumer, which is really just a test app and has no impact on the consumable code.

Reply:
No prob. I don't think I understand the full context anyway. I thought I saw two methods on the same object, and figured they could be consolidated.